geps/gep-713/index.md

```diff
@@ -3,1155 +3,1114 @@
 
-Although Direct and Inherited Policies behave differently in many respects, in
-general they should be named using similar rules.
+```yaml
+apiVersion: policies.controller.io/v1
+kind: ColorPolicy
+metadata:
+  name: my-color-policy
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io/v1
+    kind: Gateway
+    name: my-gateway ## name of the target object of Gateway kind
+  color: blue
+```
+
+<details>
+  <summary>Implementation tip</summary>
+
+  This targeting method can be implemented in Golang by using a type such as Gateway API's [`LocalPolicyTargetReference`](https://pkg.go.dev/sigs.k8s.io/gateway-api/apis/v1alpha2#LocalPolicyTargetReference) type. E.g.:
+
+  ```go
+  package color
+
+  import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+  )
 
-Policy objects MUST be clearly named so as to indicate that they are Policy
-metaresources.
+  type ColorPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
 
-The simplest way to do that is to ensure that the type's name contains the `Policy`
-string.
+    // Spec defines the desired state of the policy.
+    Spec ColorPolicySpec `json:"spec"`
 
-Implementations SHOULD use `Policy` as the last part of the names of object types
-that use this pattern.
+    // Status defines the current state of the policy.
+    Status ColorPolicyStatus `json:"status,omitempty"`
+  }
 
-If an implementation does not, then they MUST clearly document what objects
-are Policy metaresources in their documentation. Again, this is _not recommended_
-without a _very_ good reason.
+  type ColorPolicySpec struct {
+    // TargetRefs specify the targets of the policy by name.
+    // The following kinds are supported: …
+    // +listType=map
+    // +listMapKey=group
+    // +listMapKey=kind
+    // +listMapKey=name
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=16
+    TargetRefs []gatewayapiv1alpha2.LocalPolicyTargetReference `json:"targetRefs"`
 
-### Targeting Virtual Types
-In some cases (likely limited to mesh) we may want to apply policies to requests
-to external services. To accomplish this, implementations MAY choose to support
-a reference to a virtual resource type. For example:
+    // rest of the spec ("spec proper")…
+  }
+  ```
+</details>
+
+##### Label selectors
+
+The target reference includes the label selectors used to select a set of objects whose behavior to augment. E.g.:
 
 ```yaml
-apiVersion: networking.acme.io/v1alpha1
-kind: RetryPolicy
+apiVersion: policies.controller.io/v1
+kind: ColorPolicy
 metadata:
-  name: foo
+  name: my-color-policy
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io/v1
+    kind: Gateway
+    selector: ## label selectors to a set of objects of the Gateway kind
+      matchLabels:
+        env: production
```

Comment:
Would this still be a local namespace selector?

Reply:
It's really unspecified. However, given that cross-namespace refs are only defined/introduced in the next subsection, I understand it's implied that, by default, we mean labels in the same namespace, just like the definition of targeting by name before that makes no reference to namespaces.

Another way to put is: same namespace versus cross-namespace refs are orthogonal to referencing by name or by labels.

Reply:
How is it orthogonal? The spec should define whether the label selector is within the local namespace or not.